File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
       HashMap<Character, Character> isoMap = new HashMap<>();
        HashSet<Character> mapped = new HashSet<>();

        if(s.length()!=t.length()) return false;
  for (int i = 0; i < s.length(); i++) {
            char ch1 = s.charAt(i);
            char ch2 = t.charAt(i);

            if (isoMap.containsKey(ch1)) {
                if (isoMap.get(ch1) != ch2) return false;
            } else {
                if (mapped.contains(ch2)) return false;

                isoMap.put(ch1, ch2);
                mapped.add(ch2);
            }
        }

        return true;


        
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        if(s.length()!=t.length()) return false;
        unordered_map<char,char> map1;
        unordered_map<char,char> map2;

        for(int i=0;i<s.size();i++){
            char a=s[i];
            char b=t[i];
            if(map1.count(a)&&map1[a]!=b){
                return false;
            }
            if(map2.count(b)&&map2[b]!=a){
                return false;
            }

            map1[a]=b;
            map2[b]=a;
        }

        return true;

    }
};